### ISSUE TITLE
Support SparseTensor in tf.keras.Discretization layer.

### ISSUE THREAD

**Issue by brightcoder01** [tensorflow/tensorflow#37738]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): TF2.1 and TF2.2-rc0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Feature: [tf.keras.preprocessing.Discretization](https://github.com/tensorflow/tensorflow/blob/765ddddb2278551441ecaf45528898cd839df5c9/tensorflow/python/keras/layers/preprocessing/discretization.py#L33) can process the SparseTensor inputs.
Current State: This layer cannot process SparseTensor.

**Will this change the current api? How?**
No.

**Who will benefit with this feature?**
The model developer who want to discretize the value from Sparse Input. It's a common for recommendation & ranking scenario.

**Any Other info.**

**Comment by tanzhenyu** [comment]
Yep, we should probably support it. This will probably need some work and api approval. Do you have a more concrete use case?

**Comment by tanzhenyu** [comment]
(Actually on second thought, we can easily achieve this by return tf.SparseTensor(indices=input.indices, values=bucketize(input.values), dense_shape=input.dense_shape, so not much work). Understanding use cases are pretty helpful though :-)

**Comment by tanzhenyu** [comment]
@brightcoder01 gentle ping, do you have any concrete example? I intend to include it in the commit description.

**Comment by workingloong** [comment]
> @brightcoder01 gentle ping, do you have any concrete example? I intend to include it in the commit description.

The use case is that we want to discretize the numeric values with the missing value to integer ids and feed the ids into embedding.

Example, there are missing values of features in the dataset like:
|  age | height | income | price | label|
| ---- | ------ | --------| ----- | -----|
| 32   | 156     | 7800     | 100   |  0    |
| 56 | 172     |      | 130   |  1    |
| 47 |     |   6700   |    |  1    |
| 27 |   141  |   8100   |    |  0   |

We can save the table into a text file and read the value as a string when we create `tf.data.Dataset`. The output value of the `Dataset` for missing value is an empty string. For example, the batch data  from `Dataset` with size=4 is:
```python
features = {
    "age":[["32"], ["56"], ["47"], ["27"]],
    "height":[["156"], ["172"], [""], ["141"]],
    "income":[["7800"], [""], ["6700"], ["8100"]],
    "price":[["100"], ["130"], [""], [""]],
}
labels = [["0"], ["1"], ["1"], ["0"]]
```
Then we need to convert the string values to numeric values. Firstly, we must drop the empty string because we can not convert the empty to a numeric value. So we need to convert the tensor of each feature to a sparse tensor. Because some rows don't have any value after dropping empty string. Then we can use `tf.strings.to_number` to convert the string values to int.
For example, the sparse tensor of the batch for height is:
```python
tf.SparseTensor(
    index=[[0,0], [1,0], [3,0]],
    values=[156, 172, 141],
   dense_shape=(4,1)
)
```
Then, we want to discretize the height with boundaries=[140,150,160] and the result is
```python
tf.SparseTensor(
    index=[[0,0], [1,0], [3,0]],
    values=[2, 3, 1],
   dense_shape=(4,1)
)
```
And then, we will feed the sparse tensor into embedding using `tf.nn.safe_embedding_lookup_sparse`

Although, we can set a default value for the missing value to create a tensor and then cast the tensor to int. But, the embedding result of the row with missing value is not zero. If we use a sparse tensor without the missing value, the embedding result of the row will be zero. If we sum the embedding  results of multiple features, it will make difference.

**Comment by tanzhenyu** [comment]
Fixed here.
https://github.com/tensorflow/tensorflow/commit/4a441b23646965e615800872bfe72aa478e7e8d3
